src/Fizzics/level2.js

```diff
@@ -70,8 +74,17 @@ var FizzicsLevel2 = GObject.registerClass({
 
         this._codeview.setCompileResults([]);
 
+        // Block the normal notify handler that updates the code view, since we
+        // are propagating updates from the codeview to the GUI. Instead,
+        // connect a temporary handler that lets us know if anything actually
+        // did change.
         GObject.signal_handler_block(this._model, this._notifyHandler);
 
+        let guiUpdated = false;
+        this._tempHandler = this._model.connect('notify', () => {
```

Comment:
You don't need to store this in `this`, it can be a local variable. (Same below.)